nilearn/plotting/tests/test_html_stat_map.py

```diff
@@ -1,33 +1,342 @@
 import warnings
+from io import BytesIO
+import sys
 
-from nose.tools import assert_equal
 import numpy as np
+from numpy.testing import assert_warns, assert_raises
+
+from nibabel import Nifti1Image
 
 from nilearn import datasets, image
 from nilearn.plotting import html_stat_map
+from nilearn.image import new_img_like
+from ..js_plotting_utils import colorscale
 
 
-def test_encode_nii():
-    mni = datasets.load_mni152_template()
-    encoded = html_stat_map._encode_nii(mni)
-    decoded = html_stat_map._decode_nii(encoded)
-    assert np.allclose(mni.get_data(), decoded.get_data())
+def _check_html(html):
+    """ Check the presence of some expected code in the html
+    """
+    assert isinstance(html, html_stat_map.StatMapView)
+    assert "var brain =" in str(html)
+    assert "overlayImg" in str(html)
 
-    mni = image.new_img_like(mni, np.asarray(mni.get_data(), dtype='>f8'))
-    encoded = html_stat_map._encode_nii(mni)
-    decoded = html_stat_map._decode_nii(encoded)
-    assert np.allclose(mni.get_data(), decoded.get_data())
 
-    mni = image.new_img_like(mni, np.asarray(mni.get_data(), dtype='<i4'))
-    encoded = html_stat_map._encode_nii(mni)
-    decoded = html_stat_map._decode_nii(encoded)
-    assert np.allclose(mni.get_data(), decoded.get_data())
+def _assert_warnings_in(set1, set2):
+    """ Check that warnings are inside a list
+    """
+    assert set1.issubset(set2), ("the following warnings were not "
+                                 "expected: {}").format(set1.difference(set2))
 
 
-def _check_html(html):
-    assert isinstance(html, html_stat_map.StatMapView)
-    assert "var mni =" in str(html)
-    assert "var stat_map =" in str(html)
+def _simu_img(affine=np.eye(4)):
+
+    # Generate simple simulated data with one "spot"
+    data = np.zeros([8, 8, 8])
+    data[4, 4, 4] = 1
+    img = Nifti1Image(data, affine)
+    return img, data
+
+
+def _check_affine(affine):
+    # Check positive and isotropic diagonal
+    assert(affine[0, 0] == affine[1, 1])
+    assert(affine[2, 2] == affine[1, 1])
+    assert(affine[0, 0] > 0)
+
+    # Check near-diagonal affine
+    A, b = image.resampling.to_matrix_vector(affine)
+    assert(np.all((np.abs(A) > 0.001).sum(axis=0) == 1))
+
+
+def test_data2sprite():
+
+    # Generate a simulated volume with a square inside
+    data = np.zeros([8, 8, 8])
+    data[2:6, 2:6, 2:6] = 1
+
+    # turn that into a sprite and check it has the right shape
+    sprite = html_stat_map._data2sprite(data)
```

Comment:
Rename `_data2sprite` to `_data_to_sprite` since that is the convention we (and you) have followed everywhere.

Reply:
Rename `test_data2sprite` as well.

Reply:
done.